src/internal/connector/onedrive/drive.go

```diff
@@ -190,23 +194,29 @@ func collectItems(
 			)
 		}
 
-		err = collector(ctx, driveID, driveName, r.GetValue(), oldPaths, newPaths, excluded)
+		vals, err := pager.ValuesIn(page)
+		if err != nil {
+			return "", nil, nil, errors.Wrap(err, "extracting items from response")
+		}
+
+		err = collector(ctx, driveID, driveName, vals, oldPaths, newPaths, excluded)
 		if err != nil {
 			return "", nil, nil, err
 		}
 
-		if r.GetOdataDeltaLink() != nil && len(*r.GetOdataDeltaLink()) > 0 {
-			newDeltaURL = *r.GetOdataDeltaLink()
+		nextLink, deltaLink := gapi.NextAndDeltaLink(page)
+
+		if len(deltaLink) > 0 {
```

Comment:
I know the logic is from before, but graph api either gives a `nextLink` or a `deltaLink` . Don't think both will be available.

Reply:
yea, I think that's what I've read as well. At the moment I believe this pattern is used in a few different places in the codebase so I'd kind of like to keep with the pattern. If we change it, I'd suggest a separate PR that changes all instances instead of a more ad-hoc approach. After all, having consistency in handling of things like this can help with thinking through how parts of the system work